autopilot/heuristic_constraints.go

```diff
@@ -0,0 +1,121 @@
+package autopilot
+
+import (
+	"fmt"
+
+	"github.com/btcsuite/btcutil"
+)
+
+// HeuristicConstraints is a struct that indicate the constraints an autopilot
+// heuristic must adhere to when opening channels.
+type HeuristicConstraints struct {
+	// MinChanSize is the smallest channel that the autopilot agent should
+	// create.
+	MinChanSize btcutil.Amount
+
+	// MaxChanSize the largest channel that the autopilot agent should
+	// create.
+	MaxChanSize btcutil.Amount
+
+	// ChanLimit the maximum number of channels that should be created.
+	ChanLimit uint16
+
+	// Allocation the percentage of total funds that should be committed to
+	// automatic channel establishment.
+	Allocation float64
+}
+
+// moreChans returns whether we can open more channels and still be within the
+// set constraints. In the case that is possible, the funds available for new
+// channels, and the number of additional channels that can opened is also
+// returned.
+func (h *HeuristicConstraints) moreChans(channels []Channel,
```

Comment:
This was initially written with the assumption that there'd only ever be a _single_ heuristic that drove the entire agent. With this multi-heuristics scoring based model, I think we may need to re-think this approach all together.